### PR TITLE
[Common] Logging mechanism using dlog API

### PR DESCRIPTION
**Pull request by rafalgalka**
To see log messages enable platform log first:
zypper in dlogutil
dlogctrl set platformlog 1
journalctl -f

BUG=XWALK-3567

### CODE REVIEW

File: src/common/common.gypi [hunk 1]
```diff
@@ -26,7 +26,12 @@
         },
       }],
       ['tizen == 1', {
-        'defines': ['TIZEN']
+        'defines': ['TIZEN'],
+        'variables': {
+          'packages': [
+            'dlog',
+          ],
+        },
```
Comment (halton) [inline src/common/common.gypi]: I would let this new dependency protected by condition of `extension_build_type== "Debug"`

File: src/common/common.gypi [hunk 2]
```diff
@@ -68,6 +74,8 @@
     'sources': [
       'extension.cc',
       'extension.h',
+      'logger.cc',
+      'logger.h',
```
Comment (halton) [inline src/common/common.gypi]: ditto

File: src/common/logger.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#define COMPACT_LOG_INFO \
+  LogMessage(__MODULE__, __func__, __LINE__, DLOG_INFO).stream()
+#define COMPACT_LOG_WARN \
+  LogMessage(__MODULE__, __func__, __LINE__, DLOG_WARN).stream()
+#define COMPACT_LOG_ERROR \
+  LogMessage(__MODULE__, __func__, __LINE__, DLOG_ERROR).stream()
+
+#define LOGGER(priority) COMPACT_LOG_ ## priority
+#define LOGGER_IF(priority, condition) \
+  !(condition) ? (void) 0 : LogMessageVoidify() & (LOGGER(priority))
```
Comment (halton) [inline src/common/logger.h]: 4-space indent for above unfinished lines

File: src/common/logger.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Samsung Electronics Co, Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef COMMON_LOGGER_H_
+#define COMMON_LOGGER_H_
+
+#include <dlog.h>
+#include <string>
```
Comment (deqing) [inline src/common/logger.h]: Looks like `std::string` is not used here, if so you can remove it.

File: src/common/common.gypi (hunk 1 above)
Comment (rafalgalka) [inline src/common/common.gypi]: It's intended to be warning/error logger in release build. Debug messages will be ignored.

File: src/common/logger.h (hunk 3 above)
Comment (rafalgalka) [inline src/common/logger.h]: done

File: src/common/logger.h (hunk 4 above)
Comment (rafalgalka) [inline src/common/logger.h]: done

**Comment by halton** [comment]
lgtm, merging.

**Pull request merged**